cl/lib/model_helpers.py

```diff
@@ -5,10 +5,10 @@
 def make_upload_path(instance, filename):
     """Return a string like pdf/2010/08/13/foo_v._var.pdf"""
     try:
-        # Cannot do proper type checking here because of circular import
-        # problems when importing Audio, Document, etc.
-        d = instance.file_with_date
```

Comment:
This API kind of sucks, but the idea here is that before you call this function on an object, you add the `file_with_date` attribute to the object. [Here's an example of this in action][1]. The reason this is needed is because this function is used by a bunch of different objects: Audio, OpinionClusters, and eventually others, and we don't know what the correct attribute is for each of them.

I'd normally change the API so it didn't suck like this, BUT this function is [called by the model][2], and we don't get to mess with its parameters.

Maybe a better way to handle this would be to create an `@property` for every model that uses `make_upload_path` that's called something like `date_for_file_path`, and which returns the correct attribute. That'd move the logic to a better location than this hack.

[1]: https://github.com/freelawproject/courtlistener/blob/ab99e626db745eb6effc3f0e26e50b33f6706c8d/cl/scrapers/tasks.py#L405-L407
[2]: https://github.com/freelawproject/courtlistener/blob/2b1b97edf250363d92e5fd2d54f49ebe1eaf07f8/cl/audio/models.py#L83